query/src/main/java/org/infinispan/query/impl/massindex/IndexWorker.java

```diff
@@ -151,7 +153,15 @@ private Object extractValue(Object storageValue, DataConversion valueDataConvers
       @Override
       public void writeObject(ObjectOutput output, IndexWorker worker) throws IOException {
          output.writeObject(worker.cacheName);
-         output.writeObject(PojoIndexedTypeIdentifier.convertToLegacy(worker.indexedType));
+         if (worker.indexedTypes == null) {
+            output.writeInt(0);
```

Comment:
When reading, this silently leads to null being transformed into empty set. Do we care?

Reply:
not really, when MassIndexing only a set of keys, the types are ignored